I'm getting indeed
```
[INFO] Results:
[INFO] 
[INFO] Tests run: 179, Failures: 0, Errors: 0, Skipped: 0
```
but then
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:19 min
[INFO] Finished at: 2024-04-27T16:03:00+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-report-plugin:3.2.5:report-only (generate-test-report) on project glassfish-external-tck-jsonp: Execution generate-test-report of goal org.apache.maven.plugins:maven-surefire-report-plugin:3.2.5:report-only failed: Error parsing JUnit XML report appserver/tests/tck/jsonp/target/surefire-reports/TEST-ee.jakarta.tck.jsonp.api.jsongeneratortests.ClientTests.xml: An invalid XML character (Unicode: 0xffff) was found in the CDATA section.
```

```
appserver/tests/tck/jsonp/target/surefire-reports/TEST-ee.jakarta.tck.jsonp.api.jsongeneratortests.ClientTests.xml:266.26: Input is not proper UTF-8, indicate encoding !
Bytes: 0xEF 0xBF 0xBF 0x22: Bytes: 0xEF 0xBF 0xBF 0x22

sonGeneratorObjectTest5 Testing read of {"unicodechars":"\u0000\u000f\u001fÿ＀
                                                                               ^
appserver/tests/tck/jsonp/target/surefire-reports/TEST-ee.jakarta.tck.jsonp.api.jsongeneratortests.ClientTests.xml - invalid
```

(openjdk 21.0.3, Apache Maven 3.9.6)

--- a/appserver/tests/tck/jsonp/pom.xml
+++ b/appserver/tests/tck/jsonp/pom.xml
@@ -41,6 +41,9 @@
         <glassfish.version>${project.version}</glassfish.version>
         <glassfish.root>${project.build.directory}</glassfish.root>
         <glassfish.home>${glassfish.root}/glassfish7</glassfish.home>
+        <glassfish.artifact.id>glassfish</glassfish.artifact.id>
+
+        <skip.surefire.report>true</skip.surefire.report>
     </properties>
 
     <dependencyManagement>
@@ -107,7 +110,7 @@
                             <artifactItems>
                                 <artifactItem>
                                     <groupId>org.glassfish.main.distributions</groupId>
-                                    <artifactId>${glassfish-artifact-id}</artifactId>
+                                    <artifactId>${glassfish.artifact.id}</artifactId>
                                     <version>${glassfish.version}</version>
                                     <type>zip</type>
                                     <overWrite>false</overWrite>
@@ -149,6 +152,7 @@
                 <configuration>
                     <outputDirectory>${project.build.directory}/surefire-reports</outputDirectory>
                     <outputName>test-report</outputName>
+                    <skipSurefireReport>${skip.surefire.report}</skipSurefireReport>
                 </configuration>
             </plugin>
         </plugins>
@@ -156,19 +160,18 @@
 
     <profiles>
         <profile>
-            <id>full</id>
-            <activation>
-                <activeByDefault>true</activeByDefault>
-            </activation>
-            <properties>
-                <glassfish-artifact-id>glassfish</glassfish-artifact-id>
-            </properties>
-        </profile>
-
-        <profile>
             <id>web</id>
             <properties>
-                <glassfish-artifact-id>web</glassfish-artifact-id>
+                <glassfish.artifact.id>web</glassfish.artifact.id>
+            </properties>
+        </profile>
+        <profile>
+            <id>report</id>
+            <activation>
+                <jdk>[11,18)</jdk>
+            </activation>
+            <properties>
+                <skip.surefire.report>false</skip.surefire.report>
             </properties>
         </profile>
     </profiles>

--- a/appserver/tests/tck/jsonp/pom.xml
+++ b/appserver/tests/tck/jsonp/pom.xml
@@ -88,9 +88,9 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.netbeans.tools</groupId>
+            <groupId>jakarta.tck</groupId>
             <artifactId>sigtest-maven-plugin</artifactId>
-            <version>1.7</version>
+            <version>2.2</version>
         </dependency>
     </dependencies>
 

--- a/appserver/tests/tck/jsonp/pom.xml
+++ b/appserver/tests/tck/jsonp/pom.xml
@@ -150,13 +150,17 @@
                     </execution>
                 </executions>
                 <configuration>
-                    <outputDirectory>${project.build.directory}/surefire-reports</outputDirectory>
+                    <linkXRef>false</linkXRef>
                     <outputName>test-report</outputName>
                     <skipSurefireReport>${skip.surefire.report}</skipSurefireReport>
                 </configuration>
             </plugin>
         </plugins>
     </build>
+
+    <reporting>
+        <outputDirectory>${project.build.directory}/surefire-reports</outputDirectory>
+    </reporting>
 
     <profiles>
         <profile>

